web/src/screens/private/Groups/ViewGroups.js

```diff
@@ -3,25 +3,173 @@ import PropTypes from 'prop-types';
 import { compose } from 'recompose';
 import { graphql } from 'react-apollo';
 import { connect } from 'react-redux';
+import _ from 'lodash';
 
 import { withStyles } from 'material-ui/styles';
 import { CircularProgress } from 'material-ui/Progress';
+import Paper from 'material-ui/Paper';
+import Tabs, { Tab } from 'material-ui/Tabs';
+import Grid from 'material-ui/Grid';
+import Select from 'material-ui/Select';
+import { MenuItem } from 'material-ui/Menu';
+import { InputLabel } from 'material-ui/Input';
+import { FormControl } from 'material-ui/Form';
+import TextField from 'material-ui/TextField';
+
 
 import CURRENT_USER_QUERY from '../../../graphql/current-user.query';
 
 import styles from './ViewGroups.styles';
 
-const ViewGroups = ({ classes, loading }) => {
-  if (loading) {
-    return <CircularProgress className={classes.progress} size={50} />;
-  }
+import DisplayGroupsTable from '../../partial/DisplayGroupsTable';
 
-  return <div className={classes.root}>View Groups </div>;
-};
+class ViewGroups extends React.Component {
+  state = {
+    tab: 0,
+    locationFilter: '',
+    capabilityFilter: '',
+    searchFilter: '',
+  };
+
+  handleSelectChange = (e, key) => {
+    const { value } = e.target;
+    this.setState({ [key]: value });
+  };
+
+  handleTabChange = (event, tab) => {
+    this.setState({ tab });
+  };
+
+  render() {
+    const { classes, loading, user } = this.props;
+    const { tab, locationFilter, capabilityFilter, searchFilter } = this.state;
+
+    if (loading) {
+      return <CircularProgress className={classes.progress} size={50} />;
+    }
+
+    const allGroups = user.organisation.groups;
+    const myGroups = user.groups;
+    const allLocations = _.keys(_.transform(allGroups, (result, group) => {
```

Comment:
this logic is pretty hard to follow - it'd be good to toss a comment in explaining the result for future readers
```
// get all tags of type `orgStructure` from all groups in the organisation
```
(even better would be to make a function that does this)